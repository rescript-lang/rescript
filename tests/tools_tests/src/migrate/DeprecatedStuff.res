@send
external slice: (string, ~from: int, ~to_: int) => string = "slice"

@send
external shift: array<'a> => option<'a> = "shift"

module Constants = {
  let otherThing = [2, 3]
}

let deprecatedThing = [1, 2]
