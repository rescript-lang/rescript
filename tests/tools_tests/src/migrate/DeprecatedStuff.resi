@deprecated({
  reason: "Use `String.slice` instead",
  migrate: String.slice(
    ~start=%insert.labelledArgument("from"),
    ~end=%insert.labelledArgument("to_"),
  ),
})
@send
external slice: (string, ~from: int, ~to_: int) => string = "slice"

@send
@deprecated({
  reason: "Use `Array.shift` instead.",
  migrate: Array.shift(),
})
external shift: array<'a> => option<'a> = "shift"
