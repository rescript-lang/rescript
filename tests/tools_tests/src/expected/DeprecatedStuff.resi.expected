DeprecatedStuff.resi: File did not need migration
